src/simplewallet/simplewallet.cpp

```diff
@@ -8354,3 +8581,1025 @@ int main(int argc, char* argv[])
+      break;
+    }
+    case mms::message_processing::submit_tx:
+      text += tr("Submit tx");
+      break;
+    default:
+      text += tr("unknown");
+      break;
+    }
+    success_msg_writer() << text;
+  }
+
+  std::string line = input_line(tr("Choice: "));
+  if (std::cin.eof() || line.empty())
+  {
+    return false;
+  }
+  bool choice_ok = get_number_from_arg(line, choice, 1, choices);
+  if (choice_ok)
+  {
+    choice--;
+  }
+  else
+  {
+    fail_msg_writer() << tr("Wrong choice");
+  }
+  return choice_ok;
+}
+
+void simple_wallet::list_mms_messages(const std::vector<mms::message> &messages)
+{
+  success_msg_writer() << boost::format("%4s %-4s %-30s %-21s %7s %3s %-15s %-40s") % tr("Id") % tr("I/O") % tr("Coalition Member")
+          % tr("Message Type") % tr("Height") % tr("R") % tr("Message State") % tr("Since");
+  mms::message_store& ms = m_wallet->get_message_store();
+  uint64_t now = time(NULL);
+  for (size_t i = 0; i < messages.size(); ++i)
+  {
+    const mms::message &m = messages[i];
+    const mms::coalition_member &member = ms.get_member(m.member_index);
+    bool highlight = (m.state == mms::message_state::ready_to_send) || (m.state == mms::message_state::waiting);
+    message_writer(m.direction == mms::message_direction::out ? console_color_green : console_color_magenta, highlight) <<
+            boost::format("%4s %-4s %-30s %-21s %7s %3s %-15s %-40s") %
+            m.id %
+            ms.message_direction_to_string(m.direction) %
+            ms.member_to_string(member, 30) %
+            ms.message_type_to_string(m.type) %
+            m.wallet_height %
+            m.round %
+            ms.message_state_to_string(m.state) %
+            (get_human_readable_timestamp(m.modified) + ", " + get_human_readable_timespan(std::chrono::seconds(now - m.modified)) + tr(" ago"));
+  }
+}
+
+void simple_wallet::list_members(const std::vector<mms::coalition_member> &members)
+{
+  success_msg_writer() << boost::format("%2s %-20s %-s") % tr("#") % tr("Label") % tr("Transport Address");
+  success_msg_writer() << boost::format("%2s %-20s %-s") % "" % tr("Auto-Config Token") % tr("Monero Address");
+  for (size_t i = 0; i < members.size(); ++i)
+  {
+    const mms::coalition_member &member = members[i];
+    std::string label = member.label.empty() ? tr("<not set>") : member.label;
+    std::string monero_address;
+    if (member.monero_address_known)
+    {
+      monero_address = get_account_address_as_str(m_wallet->nettype(), false, member.monero_address);
+    }
+    else
+    {
+      monero_address = tr("<not set>");
+    }
+    std::string transport_address = member.transport_address.empty() ? tr("<not set>") : member.transport_address;
+    success_msg_writer() << boost::format("%2s %-20s %-s") % (i + 1) % label % transport_address;
+    success_msg_writer() << boost::format("%2s %-20s %-s") % "" % member.auto_config_token % monero_address;
+    success_msg_writer() << "";
+  }
+}
```

Comment:
So if your "keys" are my "coalition members" then yes, things get vastly more readable with this.

Note however that at least until now the MMS knows nothing about the keys as `wallet2` really handles them, and e.g. cannot display from which other members it received keys by directly looking into a "MultisigV1" or "MultisigxV1" string, or cannot yet look into a unsigned tx to list exactly who signed already.

Maybe this will become necessary at some later point. This would probably need additional supporting `wallet2`  methods however, as those structures are best seen as opaque / private.